task8/TutterApp/src/main/java/dao/PostDAOImpl.java

```diff
@@ -0,0 +1,112 @@
+package dao;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import forms.FilterForm;
+import models.Constants;
+import models.Post;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PostDAOImpl implements PostDAO {
+    private static List<Post> posts;
+
+    public PostDAOImpl() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setDateFormat(new SimpleDateFormat(Constants.DATE_FORMAT));
+
+        try {
+            posts = objectMapper.readValue(new File(Constants.PATH), new TypeReference<>() {
+            });
+        } catch (IOException e) {
+            posts = new ArrayList<>();
+
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Override
+    public Optional<Post> get(Long id) {
+        return posts.stream().filter(post -> post.getId().equals(id)).findAny();
+    }
+
+    @Override
+    public List<Post> getPage(FilterForm filterForm) {
+        return posts.stream().filter(post -> (filterForm.getAuthor() == null || filterForm.getAuthor().equals(post.getAuthor())) &&
+                (filterForm.getHashTags() == null || filterForm.getHashTags().equals(post.getHashTags())) &&
+                (filterForm.getFromDate() == null || post.getCreatedAt().after(filterForm.getFromDate())) &&
+                (filterForm.getToDate() == null || post.getCreatedAt().before(filterForm.getToDate()))).
+                sorted(Comparator.comparing(Post::getCreatedAt)).
+                skip(filterForm.getSkip()).
+                limit(filterForm.getQuantity()).
+                collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean add(Post post) {
+        if (!get(post.getId()).isPresent()) {
+            posts.add(post);
+            saveToFile();
```

Comment:
What if `saveToFile` fails due to an `IOException`?) You will return true as if it is allright but it is not: post not added at all.
I would rather make `saveToFile` return true or false depending on the outcome, and in this method just `return saveToFile(..)`

Another option is to make `saveToFile` throw an exception (using `throws`) and then put `try ..catch` to the curent method
Yet first option seems better